src/components/Encounter/CreateEncounterForm.tsx

```diff
@@ -199,6 +212,14 @@ export default function CreateEncounterForm({
                             field.onChange(updatedDate.toISOString());
                           }}
                           className="h-9"
+                          disabled={(date) => {
+                            if (status !== EncounterStatus.PLANNED) {
+                              const today = new Date();
+                              today.setHours(23, 59, 59, 999);
+                              return date > today;
+                            }
+                            return false;
+                          }}
```

Comment:
code is hard to read

```suggestion
                          disabled={(date) => {
                            if (status === EncounterStatus.PLANNED) {
                              return false;
                            }
                            return date < endOfDay(new Date());
                          }}
```